bash_scripts/run_sim.sh

```diff
@@ -1,40 +1,68 @@
 
-# This script is designed to run a non-artificial intelligence simulation on the Unity cluster at UMass Amherst.
-# Users can provide custom parameters via the command line using -- before the parameter list.
+set -e  # Stop the script if any command fails
 
-# Ensure the script stops if any command fails
-set -e
+echo "Starting job with SLURM_ARRAY_TASK_ID: $SLURM_ARRAY_TASK_ID"
 
-# Change to the home directory and then to the SDN simulator directory
-# shellcheck disable=SC2164
-cd
-# shellcheck disable=SC2164
-cd /work/pi_vinod_vokkarane_uml_edu/git/sdn_simulator/
+# Load the Python 3.11.7 module
+module load python/3.11.7
 
-# Load the required Python module
-module load python/3.11.0
+# Verify the loaded Python version
+python --version
 
-# Activate the virtual environment
-source venvs/unity_venv/venv/bin/activate
+# Define parameter arrays
+allocation_methods=("first_fit" "last_fit")
+spectrum_allocation_priorities=("CSB" "BSC")
+cores_per_link=(4)
 
-# Install the required Python packages
-pip install -r requirements.txt
+# Define additional variables
+num_requests=15000
+multi_fiber=False
 
-# Register custom reinforcement learning environments in Stable Baselines 3
-./bash_scripts/register_rl_env.sh ppo SimEnv
+# Calculate total combinations
+total_combinations=$(( ${#allocation_methods[@]} * ${#spectrum_allocation_priorities[@]} * ${#cores_per_link[@]} ))
 
-# Define default parameters for the simulation
-NETWORK="Pan-European"
-TRAIN_FILE_PATH="Pan-European/0531/22_00_16_630834"
-ML_MODEL="knn"
+# Ensure the task ID is within the range of total combinations
+if [ "$SLURM_ARRAY_TASK_ID" -ge "$total_combinations" ]; then
+  echo "SLURM_ARRAY_TASK_ID out of range."
+  exit 1
+fi
 
-# Run the non-artificial intelligence simulation with the specified parameters
-python run_sim.py --network "$NETWORK" --train_file_path "$TRAIN_FILE_PATH" --ml_model "$ML_MODEL"
+# Calculate indices based on SLURM_ARRAY_TASK_ID
+allocation_method_index=$(( SLURM_ARRAY_TASK_ID / (${#spectrum_allocation_priorities[@]} * ${#cores_per_link[@]}) ))
+temp_index=$(( SLURM_ARRAY_TASK_ID % (${#spectrum_allocation_priorities[@]} * ${#cores_per_link[@]}) ))
+spectrum_priority_index=$(( temp_index / ${#cores_per_link[@]} ))
+cores_per_link_index=$(( temp_index % ${#cores_per_link[@]} ))
+
+# Select parameters for this task
+allocation_method=${allocation_methods[$allocation_method_index]}
+spectrum_priority=${spectrum_allocation_priorities[$spectrum_priority_index]}
+cores=${cores_per_link[$cores_per_link_index]}
+
+# Print parameters for this task
+echo "Running simulation with:"
+echo "  Allocation method: $allocation_method"
+echo "  Spectrum priority: $spectrum_priority"
+echo "  Cores per link: $cores"
+echo "  Number of requests: $num_requests"
+echo "  Multi-fiber: $multi_fiber"
+
+# Move to the project directory
+cd ~/Git/SDON_simulator/ || { echo "Failed to change to SDON_simulator directory"; exit 1; }
```

Comment:
My comment is related above we should have this as input but also have a default option.